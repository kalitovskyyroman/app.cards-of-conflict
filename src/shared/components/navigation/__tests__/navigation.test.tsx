import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Navigation, { IButton } from '../navigation';

const testIdLeft = 'left-btn';
const testIdRight = 'right-btn';

const leftButtonTitleMock = 'left';
const rightButtonTitleMock = 'right';

const leftButton: IButton = { title: leftButtonTitleMock };
const rightButton: IButton = { title: rightButtonTitleMock };

describe('Navigation component', () => {
    it('render component', () => {
        const { container } = render(<Navigation leftButton={leftButton} rightButton={rightButton} />);
        expect(container).toMatchSnapshot();
    });

    it('render titles', () => {
        render(<Navigation leftButton={leftButton} rightButton={rightButton} />);

        expect(screen.getByTestId(testIdLeft)).toHaveTextContent(leftButtonTitleMock);
        expect(screen.getByTestId(testIdRight)).toHaveTextContent(rightButtonTitleMock);
    });

    it('test clicks', () => {
        const handleLeftClick = jest.fn();
        const handleRightClick = jest.fn();

        leftButton.onClick = handleLeftClick;
        rightButton.onClick = handleRightClick;

        render(<Navigation leftButton={leftButton} rightButton={rightButton} />);

        const leftBtn = screen.getByTestId(testIdLeft);
        const rightBtn = screen.getByTestId(testIdRight);

        expect(handleLeftClick).toBeCalledTimes(0);
        expect(handleLeftClick).toBeCalledTimes(0);

        fireEvent.click(leftBtn);
        expect(handleLeftClick).toBeCalledTimes(1);

        fireEvent.click(rightBtn);
        expect(handleRightClick).toBeCalledTimes(1);
    });
});
