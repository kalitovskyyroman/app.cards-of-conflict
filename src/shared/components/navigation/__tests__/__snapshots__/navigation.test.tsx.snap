// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component render component 1`] = `
<div>
  <div
    class="container"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth button css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
      data-testid="left-btn"
      tabindex="0"
      type="button"
    >
      left
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth button css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
      data-testid="right-btn"
      tabindex="0"
      type="button"
    >
      right
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
